inst/stan/epinowcast.stan

```diff
@@ -22,78 +22,128 @@ data {
-  // Priors (1st index = mean, 2nd index = standard deviation)
-  array[2] real eobs_lsd_p; // Standard deviation for expected final observations
-  array[2] real refp_mean_int_p; // log mean intercept for reference date delay
-  array[2] real refp_sd_int_p; // log standard deviation for the reference date delay
-  array[2] real refp_mean_beta_sd_p; // standard deviation of scaled pooled logmean effects
-  array[2] real refp_sd_beta_sd_p; // standard deviation of scaled pooled logsd effects
-  array[2] real rep_beta_sd_p; //standard deviation of scaled pooled report date effects
-  array[2] real sqrt_phi_p; // 1/sqrt(overdispersion)
 }
 
 transformed data{
+  array[g] int groups;
   real logdmax = 5*log(dmax); // scaled maxmimum delay to log for crude bounds
   // prior mean of cases based on thoose observed
   vector[g] eobs_init = log(to_vector(latest_obs[1, 1:g]) + 1);
-  // if no reporting day effects use native probability for reference day
-  // effects
+  // if no reporting day effects use probability for reference day effects
   int ref_as_p = (model_rep > 0 || model_refp > 0) ? 0 : 1; 
+  for (i in 1:g) {
+    groups[i] = i;
+  }
 }
 
 parameters {
+  // Expectation model
   array[g] real leobs_init; // First time point for expected observations
   vector<lower=0>[g] eobs_lsd; // standard deviation of rw for primary obs
   array[g] vector[t - 1] leobs_resids; // unscaled rw for primary obs
-  array[model_refp ? 1 : 0] real<lower=-10, upper=logdmax> refp_mean_int; // logmean intercept
-  array[model_refp > 1 ? 1 : 0]real<lower=1e-3, upper=2*dmax> refp_sd_int; // logsd intercept
-  vector[model_refp ? refp_fncol : 0] refp_mean_beta; // unscaled modifiers to log mean
-  vector[model_refp > 1 ? refp_fncol : 0] refp_sd_beta; // unscaled modifiers to log sd
-  vector[rep_fncol] rep_beta; // unscaled modifiers to report date hazard
-  vector<lower=0>[refp_rncol] refp_mean_beta_sd; // pooled modifiers to logmean
-  vector<lower=0>[model_refp ? refp_rncol : 0] refp_sd_beta_sd; // pooled modifiers to logsd
-  vector<lower=0>[rep_rncol] rep_beta_sd; // pooled modifiers to report date
-  array[model_obs > 0 ? 1 : 0] real<lower=0> sqrt_phi; // Overall dispersion by group
+
+  // Reference model
+  array[model_refp ? 1 : 0] real<lower=-10, upper=logdmax> refp_mean_int;
+  array[model_refp > 1 ? 1 : 0]real<lower=1e-3, upper=2*dmax> refp_sd_int; 
+  vector[model_refp ? refp_fncol : 0] refp_mean_beta; 
+  vector[model_refp > 1 ? refp_fncol : 0] refp_sd_beta; 
+  vector<lower=0>[refp_rncol] refp_mean_beta_sd;
+  vector<lower=0>[model_refp ? refp_rncol : 0] refp_sd_beta_sd; 
+
+  // Report model
+  vector[rep_fncol] rep_beta;
+  vector<lower=0>[rep_rncol] rep_beta_sd; 
+
+  // Missing reference date model
+  array[model_miss] real miss_int;
+  vector[miss_fncol] miss_beta; 
+  vector<lower=0>[miss_rncol] miss_beta_sd; 
+
+  // Observation model
+  array[model_obs > 0 ? 1 : 0] real<lower=0> sqrt_phi; // Overall dispersion
```

Comment:
I wanted to be able to avoid forcing an intercept but ultimately couldn't. Maybe something for the future if its something people actually want control of (in most cases I think this as is likely fine)